File: 2016/luke07.hs
main :: IO ()
main = do desc <- lines <$> readFile "luke07_data.txt"
          print . direction $ words <$> desc

direction :: [[String]] -> (Int, Int)
direction = foldl add (0,0)
    where add (n,w) [_, m, _, dir] = case dir of
            "north" -> (n + read m, w)
            "south" -> (n - read m, w)
            "west"  -> (n, w + read m)
            _       -> (n, w - read m)

File: 2016/luke07.rb
p File.readlines('luke07_data.txt').inject([0,0]){ |res, line|
  _, m, _, dir = line.strip.split(/\s+/)
  res[0] += m.to_i if dir == "north"
  res[0] -= m.to_i if dir == "south"
  res[1] += m.to_i if dir == "west"
  res[1] -= m.to_i if dir == "east"
  res
}
